### PR TITLE
drawer details component and redux store

### PR DESCRIPTION
**Pull request by DeAlexandra**
-added drawer details component 
-created two form components for "transactions" and "users" details
-created redux store, action and reducer for the details drawer state management
-added user detail card and table entry handlers to change the state

_Need to work on the conditional rendering of the form in the "DrawerDetails" component._

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+import store from "./context/redux/store";
 import DrawerMenu from "./components/DrawerMenu";
 import Header from "./components/Header";
 import Footer from "./components/Footer";
 import ToastContext from "./context/ToastContext";
 import ToastNotification from "./components/Toast alerts/ToastNotification";
-import TransactionDetails from "./pages/transactions/TransactionDetails";
+import DrawerDetails from "./components/DrawerDetails";
 
 const Transactions = lazy(() => import("./pages/transactions/Transactions"));
```
Comment (enfrbhd) [inline src/App.js:13]: please use prefetching strategy for lazy loading with webpack comments (that strategy enables the prefetching of the files and caching them before they are requested on a lazy loaded route (the timing of fetching of a component from cache will be of 1-2ms)

File: src/components/users/TransactionForm.jsx [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
 
-export default function TransactionDetails() {
+export default function TransactionForm() {
 
     const [singleTransaction, setSingleTransaction] = useState([]);
     const [isLoading, setIsLoading] = useState(false);
     const { setToastState } = useContext(ToastContext);
     const navigate = useNavigate();
-    const { id } = useParams();
+    const location = useLocation();
```
Comment (enfrbhd) [inline src/components/users/TransactionForm.jsx:21]: very nice :)

File: src/components/users/UserForm.jsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+export default function UserForm() {
+    const [singleUser, setSingleUser] = useState([]);
+    const { t } = useTranslation();
+    const [isLoading, setIsLoading] = useState(false);
+    const { setToastState } = useContext(ToastContext);
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const getUserId = () => {
+        return (location.pathname).slice(7);
```
Comment (enfrbhd) [inline src/components/users/UserForm.jsx:22]: 7 is highly specific to the position. split the strings by "/" and take the last element of that array

File: src/components/users/UserForm.jsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    };
+
+    const userId = getUserId();
+
+    useEffect(() => {
+        fetchUser();
+    }, [userId]);
+
+    const fetchUser = async () => {
```
Comment (enfrbhd) [inline src/components/users/UserForm.jsx:31]: fetchUser and updateUser can sit outside the component itself in order not to clutter the component. you can send them state parameters and callback parameters from this component, but their definition can sit outside of it

File: src/context/redux/actions.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+
+export const setOpen = () => ({ type: "OPEN" });
```
Comment (enfrbhd) [inline src/context/redux/actions.js:2]: actions can be simple objects that have a type. adding a wrapper function is unnecessary.

File: src/components/users/UserForm.jsx (hunk 3 above)
Comment (DeAlexandra) [reply src/components/users/UserForm.jsx]: changed the function to get the last element of the array

File: src/context/redux/actions.js (hunk 5 above)
Comment (DeAlexandra) [reply src/context/redux/actions.js]: removed wrapper function

**Review by enfrbhd — APPROVED** [review]

**Pull request merged**